tests/TestPsci/EvalTest.hs

```diff
@@ -0,0 +1,67 @@
+module TestPsci.EvalTest where
+
+import Prelude ()
+import Prelude.Compat
+
+import           Control.Monad (forM_, foldM_)
+import           Control.Monad.IO.Class (liftIO)
+import           Data.List (isPrefixOf, intercalate)
+import           Data.List.Split (splitOn)
+import           System.Directory (getCurrentDirectory)
+import           System.Exit (exitFailure)
+import           System.FilePath ((</>), takeFileName)
+import qualified System.FilePath.Glob as Glob
+import           System.IO.UTF8 (readUTF8File)
+import           Test.HUnit
+import           TestPsci.TestEnv
+
+evalTests :: Test
+evalTests = TestLabel "evalTests" . TestCase $ do
+  putStrLn "\nPSCi eval test suites"
+  testFiles <- evalTestFiles
+  forM_ testFiles evalTest
+
+evalTestFiles :: IO [FilePath]
+evalTestFiles = do
+  cwd <- getCurrentDirectory
+  let psciExamples = cwd </> "examples" </> "psci"
+  Glob.globDir1 (Glob.compile "**/*.purs") psciExamples
+
+data EvalLine = Line String
```

Comment:
I've once tried to use `Text`, but many functions used in the test is just using `String`, and I though if we use `Text`, there will be lots of `T.pack` and `T.unpack` everywhere.

For example, in PSCi tests and PSCi itself uses `readProcessWithExitCode` to run JS code in Node.js, and it just returns `String`. Also, `parseCommand` of PSCi parser is using `String` as its input. The lines will be evaluated and compared with results often, so I think `String` will work here.